python/samples/concepts/memory/new_memory.py

```diff
@@ -162,16 +170,23 @@ class DataModelList:
         collection_name=collection_name,
         create_database=True,
     ),
+    "azure_cosmos_mongodb": lambda: AzureCosmosDBforMongoDBCollection(
+        data_model_type=DataModel,
+        collection_name=collection_name,
+    ),
 }
 
 
 def print_record(result: VectorSearchResult | None = None, record: DataModel | None = None):
     if result:
         record = result.record
-    print(f"  Found id: {record.id}")
-    print(f"    Content: {record.content}")
+    print(Colors.CGREEN + f"  Found id: {record.id}" + Colors.CEND)
+    if result and result.score is not None:
+        print(Colors.CGREEN + f"    Score: {result.score}" + Colors.CEND)
+    print(Colors.CWHITE + f"    Content: {record.content}")
+    print(f"    Tag: {record.tag}" + Colors.CEND)
     if record.vector is not None:
-        print(f"    Vector (first five): {record.vector[:5]}")
+        print(Colors.CWHITE + f"    Vector (first five): {record.vector[:5]}" + Colors.CEND)
 
 
 async def main(collection: str, use_azure_openai: bool, embedding_model: str):
```

Comment:
Similar to what @TaoChenOSU mentioned about developer "quality of life," it is my opinion that this sample has gotten pretty large and difficult to follow. I may use the word "daunting." :) Is there _any_ way we can simplify things here? What are the "easiest" components to communicate first, then medium level components, then more advanced components?

Reply:
thanks for these suggestions both! I've restructured the samples a bit to make this clearer and removed some samples with the old code!